govc/task/recent.go

```diff
@@ -101,7 +124,112 @@ func taskName(info *types.TaskInfo) string {
 	}
 }
 
+type history struct {
+	*task.HistoryCollector
+
+	cmd *recent
+}
+
+func (h *history) Collect(ctx context.Context, f func([]types.TaskInfo)) error {
+	for {
+		tasks, err := h.ReadNextTasks(ctx, 10)
+		if err != nil {
+			return err
+		}
+
+		if len(tasks) == 0 {
+			if h.cmd.follow {
+				// TODO: this only follows new events.
+				// need to watch TaskHistoryCollector.LatestPage for updates to existing Tasks
+				time.Sleep(time.Second)
+				continue
+			}
+			break
+		}
+
+		f(tasks)
+	}
+	return nil
+}
+
+type collector interface {
+	Collect(context.Context, func([]types.TaskInfo)) error
+	Destroy(context.Context) error
+}
+
+// useRecent returns true if any options are specified that require use of TaskHistoryCollector
+func (cmd *recent) useRecent() bool {
+	return cmd.begin == 0 && cmd.end == 0 && !cmd.r && len(cmd.state) == 0
+}
+
+func (cmd *recent) newCollector(ctx context.Context, c *vim25.Client, ref *types.ManagedObjectReference) (collector, error) {
+	if cmd.useRecent() {
+		// original flavor of this command that uses `RecentTask` instead of `TaskHistoryCollector`
```

Comment:
Should we document somewhere that the default behavior could fail if the number of allowed task collectors is exceeded? See https://vdc-download.vmware.com/vmwb-repository/dcr-public/184bb3ba-6fa8-4574-a767-d0c96e2a38f4/ba9422ef-405c-47dd-8553-e11b619185b2/SDK/vsphere-ws/docs/ReferenceGuide/vim.TaskManager.html#createCollector

Reply:
We `Destroy()` the collector when the command is done. The default VC max is 32, same as EventHistoryCollector - used by the govc `events` command, have not had any reports of hitting the max.